e2e/cypress/integration/api-role-access.js

```diff
@@ -0,0 +1,265 @@
+describe('Restrict access to endpoint by role', function () {
+
+  let moment = require('moment-timezone');
+  require('moment/locale/en-gb');
+  moment.locale("en-gb");
+  const todayDateString = moment().format("YYYY-MM-DD");
+  const todayString = todayDateString + "T00:00:00Z";
+  const millis = moment(todayString).unix() * 1000;
+
+  const testCases = [
+    {
+      roles: ["test"],
+      endpoint: "/airport-config",
+      method: "GET",
+      responseCode: 200
+    },
+    {
+      roles: ["test", "arrivals-and-splits:view"],
+      endpoint: "/airport-info",
+      method: "GET",
+      responseCode: 200
+    },
+    {
+      roles: ["test"],
+      endpoint: "/airport-info",
+      method: "GET",
+      responseCode: 401
+    },
+    {
+      roles: ["test"],
+      endpoint: "/alerts/" + millis,
+      method: "GET",
+      responseCode: 200
+    },
+    {
+      roles: ["test"],
+      endpoint: "/version",
+      method: "GET",
+      responseCode: 200
+    },
+    {
+      roles: ["test"],
+      endpoint: "/feed-statuses",
+      method: "GET",
+      responseCode: 200
+    },
+    {
+      roles: ["test"],
+      endpoint: "/logged-in",
+      method: "GET",
+      responseCode: 200
+    },
+    {
+      roles: [],
+      endpoint: "/data/login",
+      method: "POST",
+      responseCode: 400
+    },
+    {
+      roles: ["test"],
+      endpoint: "/data/user",
+      method: "GET",
+      responseCode: 200
+    },
+    {
+      roles: ["test"],
+      endpoint: "/logging",
+      method: "POST",
+      responseCode: 200
+    },
+    {
+      roles: ["test"],
+      endpoint: "/data/staff",
+      method: "POST",
+      responseCode: 401
+    },
+    {
+      roles: ["test", "port-operator-staff"],
+      endpoint: "/data/feed/live/lhr",
+      method: "POST",
+      responseCode: 400
+    },
+    {
+      roles: ["test"],
+      endpoint: "/data/feed/live/lhr",
+      method: "POST",
+      responseCode: 401
+    },
+    {
+      roles: [],
+      endpoint: "/logging",
+      method: "POST",
+      responseCode: 401
+    },
+    {
+      roles: ["test"],
+      endpoint: "/data/user/has-port-access",
+      method: "GET",
+      responseCode: 200
+    },
+    {
+      roles: [],
+      endpoint: "/data/user/has-port-access",
+      method: "GET",
+      responseCode: 401
+    },
+    {
+      roles: ["test"],
+      endpoint: "/alerts",
+      method: "POST",
+      responseCode: 401
+    },
+    {
+      roles: ["test"],
+      endpoint: "/alerts",
+      method: "DELETE",
+      responseCode: 401
+    },
+    {
+      roles: ["test", "fixed-points:view"],
+      endpoint: "/fixed-points",
+      method: "GET",
+      responseCode: 200
+    },
+    {
+      roles: ["test"],
+      endpoint: "/fixed-points",
+      method: "GET",
+      responseCode: 401
+    },
+    {
+      roles: ["test", "fixed-points:edit"],
```

Comment:
is this right? 400 when you have editing rights for fixed points?

Reply:
the purpose of this test is just to test permissions, so getting a 400 means that you have the relevant role - I decided it wasn't worth adding the complexity of making all these tests post valid data as well, especially as some of them are still autowired.